Libraries/ReactNative/requireNativeComponent.js

```diff
@@ -70,13 +70,19 @@ function requireNativeComponent(
     viewConfig.propTypes = null;
   }
 
-  // The ViewConfig doesn't contain any props inherited from the view manager's
-  // superclass, so we manually merge in the RCTView ones. Other inheritance
-  // patterns are currenty not supported.
-  const nativeProps = {
-    ...UIManager.RCTView.NativeProps,
-    ...viewConfig.NativeProps,
-  };
+  let baseModuleName = viewConfig.baseModuleName;
```

Comment:
`do { } while ();` would make it execute the statement at least once, which we don't want to do if there is no `baseModuleName` exported. So not sure if that is a good idea.